[TM Ready] Monkey can have a little Donut - Return of Donutstation

williamcll opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

# About The Pull Request
Brings back DonutStation, an old TG map now updated to bee standards.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

Variety good, stop being addicted to metastation
## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

![image](https://user-images.githubusercontent.com/43061493/166740862-c7565578-4eb0-45ee-b9de-2dfba58f2315.png)
The full map

![image](https://user-images.githubusercontent.com/43061493/166740915-3e3aa1af-d960-4441-a8c0-ece5de3748b1.png)
New Toxins Asteroid

![image](https://user-images.githubusercontent.com/43061493/166740978-24a1ce19-fa67-4b7f-a93d-7885bb87ebc9.png)
Expanded holodeck, Added Exploration office and science shuttle.

![image](https://user-images.githubusercontent.com/43061493/166741084-c17ef833-bcd0-410f-a685-63415e1639ea.png)
Cleaner and updated Atmospherics

![image](https://user-images.githubusercontent.com/43061493/166741201-b614c6d0-a723-4bc2-b244-0bef17b8d9b1.png)
Expanded Perma
</details>

## Changelog

:cl:
add: DonutStation
add: DonutStation Exploration office
tweak: Updates Mining vendors
del: incompatible pianos
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

williamcll commented:
TODO:
Add Exploration Office
Add Science Dock
Update Mining
Update Labour Shuttle
Update Pipe Layer
Update missing wiring

polygoblyn commented:
Got any pics?

nednaZ commented:
Linters are failing.

williamcll commented:
> Linters are failing.

I don't know what's causing the issue, all the directories are in the right folder and there's no _maps in the _maps folder.

nednaZ commented:
One thing I'll point out, you didn't create a branch for this.
I suggest creating new branches for every PR.

nednaZ commented on code/datums/shuttles.dm:
```diff
@@ -332,6 +332,13 @@
+	suffix = "donut"
+	name = "Donut Station Emergency shuttle"
+	description = "A streamlined shuttle for a streamlined getaway from whatever hells going on."
+	admin_notes = "This shuttle hasn't updated since 2018, so there might be bugs."
```
Has this shuttle been updated for this PR?

williamcll replied:
I added that note myself, walked around and I can't see any issues. Besides, this shuttle operates completely fine on NSV as well.

nednaZ reviewed (commented):
So some immediate thoughts on the station design.

The vault is in a very weird place, placed in a maint tunnel directly next to security.
The vault also should not have turrets, no other map has that.

The supermatter, by being in the dead center, makes actually getting the SM shard objective far harder as there is no easy way to sneak in.

Permabrig is entirely impossible to escape via spacewalking.

Why is AI Upload in maintenance of all places?

The Warden's office needs Captain, Walter needs to be relocated to the Lawyer's Office.

Virology only has one lab setup, despite two spawns.

Exploration dock needs some kind of guideline to show new arrivals where it is.

Overall a functional, but INCREDIBLY weird map.
I don't get why some rooms are seemingly randomly sized, some are huge while others are tiny.

williamcll commented:
Vault I can move it closer to the bridge

I can add another entrance like that of deltastation

Since when was perma possible to escape anyway? 

Blame the past people that designed this. Putting it in where the vault is now should be fine?

oversight, fill fix.

That I'm not sure how to do.

Mind you I didn't design this map, this was pulled from austation's upstream of the bee map from 2018

williamcll commented:
Changed the following:
Added oxygen and firefighting locker around the primary hall
Updated solars APCs and naming
Added a green line guide from science entrance to exploration office
Added exploration vendor
Added exploration comms in telecomms.
Changed more fixed maints into random maints.
Added more entrances to Supermatter engine room.
Update Warden and Lawyer Pets.
Added one more setup for Virology.
Cleaned up atmos piping, no more pumps under doors.
Simplified Vault Access and removed turrets.
Changed AI Upload direction

nednaZ reviewed (commented):
Being test merged.

nednaZ commented:
Second testmerge being done.

nednaZ reviewed (approved):
Looks good to me.
I can't see any further issues that need immediate fixing, only long-term play will show issues that will need fixes
